tests/test_numpy_array.cpp

```diff
@@ -405,6 +405,19 @@ TEST_SUBMODULE(numpy_array, sm) {
         return a;
     });
 
+    sm.def("array_reshape1", [](py::array_t<double> a, size_t N) {
+        return a.reshape({N, N, N});
+    });
```

Comment:
To keep the tests lean, I'd remove this function completely (and `test_array_reshape` with it). The exact same interface is covered by `test_array_reshape`. (I realize it takes away from PyPy coverage because of the `xfail` for `test_array_reshape`, but that's not enough reason in my mind to keep this one.)
If you feel differently, I'd at least change this to something like `{N, N+5, N+7}` to get a little bit more value.